boring/Cargo.toml

```diff
@@ -19,9 +19,19 @@ rustdoc-args = ["--cfg", "docsrs"]
 [features]
 # Controlling the build
 
-# Use a FIPS-validated version of boringssl.
+# Use a FIPS-validated version of BoringSSL. This feature sets "fips-compat".
 fips = ["fips-compat", "boring-sys/fips"]
 
+# Use a FIPS build of BoringSSL, but don't set "fips-compat".
+#
+# As of boringSSL commit a430310d6563c0734ddafca7731570dfb683dc19, we no longer
+# need to make exceptions for the types of BufLen, ProtosLen, and ValueLen,
+# which means the "fips-compat" feature is no longer needed.
+#
+# TODO(cjpatton) Delete this feature and modify "fips" so that it doesn't imply
+# "fips-compat".
+fips-no-compat = ["boring-sys/fips"]
```

Comment:
Note: None of the testes in CI exercise this feaature, though I've validated it in the internal oxy repo.

I could add tests, but we'd also need to bump `boring-sys/deps/boringssl-fips`, which is used to validate older builds. We could add another boringSSL submodule, but that would be a little annoying.

Reply:
You could make a test that pulls a recent version of bssl to build bindings with, but that seems a bit arduous. The change isn't particularly scary, so I wouldn't say it's strictly necessary. If things pass internally then that's good. When we cut 5.0 we'll make sure CI is up to date